### PR TITLE
test: fix jsonschema version checking in pro test

### PR DESCRIPTION
**Pull request by TheRealFalcon**
## Proposed Commit Message

```
test: fix jsonschema version checking in pro test
```

## Additional Context
https://launchpadlibrarian.net/715151681/buildlog_ubuntu-focal-amd64.cloud-init_23.4.daily-202402191902-4779679e~ubuntu20.04.1_BUILDING.txt.gz

https://launchpadlibrarian.net/715105971/buildlog_ubuntu-jammy-amd64.cloud-init_23.4.daily-202402191436-4779679e~ubuntu22.04.1_BUILDING.txt.gz

## Checklist

- [ ] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/development/index.html)
- [ ] I have updated or added any [unit tests](https://cloudinit.readthedocs.io/en/latest/development/testing.html) accordingly
- [ ] I have updated or added any [documentation](https://cloudinit.readthedocs.io/en/latest/development/contribute_docs.html) accordingly

## Merge type

- [x] Squash merge using "Proposed Commit Message"
- [ ] Rebase and merge unique commits. Requires commit messages per-commit each referencing the pull request number (#<PR_NUM>)

### CODE REVIEW

File: tests/unittests/config/test_cc_ubuntu_pro.py [hunk 1]
```diff
@@ -443,8 +444,12 @@ class TestUbuntuProSchema:
                         " Use ``ubuntu_pro`` instead"
                     ),
                 ),
+                # If __version__ no longer exists on jsonschema, that means
```
Comment (blackboxsw) [inline tests/unittests/config/test_cc_ubuntu_pro.py:447]: While this is a nice approach, and future-proof for when __version__ disappears. We still get a warning from pytest -[about the deprecation of __version__ in jsonschema](https://github.com/python-jsonschema/jsonschema/blob/main/jsonschema/__init__.py#L29-L34). Let's instead heed their deprecation warning advice.

WDYT about the following tested on focal lowest-supported 2.6.0 and py3 4.21:
```diff
--- a/tests/unittests/config/test_cc_ubuntu_pro.py
+++ b/tests/unittests/config/test_cc_ubuntu_pro.py
@@ -430,6 +430,12 @@ JSONSCHEMA_SKIP_REASON = (
 )
 
 
+def getJSONSchemaVersion() -> Version:
+    from importlib import metadata
+
+    return Version.from_str(metadata.version("jsonschema"))
+
+
 class TestUbuntuProSchema:
     @pytest.mark.parametrize(
         "config, expectation, skip_reason",
@@ -448,8 +454,7 @@ class TestUbuntuProSchema:
                 # we're using a high enough version of jsonschema to not need
                 # to skip this test.
                 JSONSCHEMA_SKIP_REASON
-                if Version.from_str(getattr(jsonschema, "__version__", "999"))
-                < Version(4)
+                if getJSONSchemaVersion() < Version(4)
                 else "",
                 id="deprecation_of_ubuntu_advantage_skip_old_json",
             ),
```

**Review by blackboxsw — CHANGES_REQUESTED** [review]

File: tests/unittests/config/test_cc_ubuntu_pro.py (hunk 1 above)
Comment (TheRealFalcon) [reply tests/unittests/config/test_cc_ubuntu_pro.py:447]: We're still supporting 3.6 and 3.7, and this solution doesn't work there.

Comment (blackboxsw) [reply tests/unittests/config/test_cc_ubuntu_pro.py:447]: Sure enough `New in version 3.8.` good catch. Hadn't thought through that as focal is 3.8

**Review by blackboxsw — APPROVED** [review]
Given python 3.6/3.7 support needs +1 on your suggestion.

**Pull request merged**